### PR TITLE
chore: update ubuntu and add safe dir vars

### PR DESCRIPTION
**Pull request by ChiefHolland**
# Contributor Comments

This PR updates the base Ubuntu image to 22.04 and adds environment variables for detecting a .git directory mounted in the container and setting it as safe. 

## Pull Request Checklist

Thank you for submitting a contribution to `easy_infra`!

In order to streamline the review of your contribution we ask that you review and comply with the below requirements:

- [x] Rebase your branch against the latest commit of the target branch
- [x] If you are adding a dependency, please explain how it was chosen
- [x] If manual testing is needed in order to validate the changes, provide a testing plan and the expected results
- [x] If there is an issue associated with your Pull Request, link the issue to the PR.

### CODE REVIEW

**Comment by JonZeolla** [comment]
https://github.com/SeisoLLC/easy_infra/blob/cb54751d9ff5e6bb7eccd8211bbbedfc9c341335/easy_infra/utils.py#L41 may be the reason for the test failure. It's doing updates using a previously release `easy_infra` image which is 20.04 so it won't align with 22.04.  Probably need to manually fix (find/specify the right version for 22.04 in `easy_infra.yml`) and then after this merge the auto updates will work again.

File: CONTRIBUTING.md
Comment (JonZeolla) [inline CONTRIBUTING.md:69]: Let's use a `GIT_SAFE_DIRECTORY` and then map it into `GIT_CONFIG_VALUE_0`

Comment (JonZeolla) [reply CONTRIBUTING.md:69]: Also, move these to the right place in `docs/`

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:93]: Please include specific pointers to the minimum version needed and the git commit on git/git explaining why

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:96]: tar extract to the `git/` dir, and change `cd` on the next line to `cd git`

File: build/docker-entrypoint.sh
Comment (JonZeolla) [inline build/docker-entrypoint.sh:43]:
```suggestion
if [[ -n ${GIT_SAFE_DIRECTORY:-} ]]; then
```

File: build/docker-entrypoint.sh
Comment (JonZeolla) [inline build/docker-entrypoint.sh:45]:
```suggestion
  GIT_CONFIG_VALUE_0="${GIT_SAFE_DIRECTORY}"
```

File: build/docker-entrypoint.sh
Comment (JonZeolla) [inline build/docker-entrypoint.sh:44]:
```suggestion
  # Let GIT_SAFE_DIRECTORY override the default safe directory location
```

File: easy_infra.yml
Comment (JonZeolla) [inline easy_infra.yml:34]: Make sure to test `task update`

**Review by JonZeolla — CHANGES_REQUESTED** [review]

File: easy_infra.yml
Comment (ChiefHolland) [reply easy_infra.yml:34]: task update functioned after my changes above. Although it reset the two version changes here back to their original values.

**Comment by JonZeolla** [comment]
This is blocked on https://github.com/SeisoLLC/easy_infra/pull/279

File: Taskfile.yml
Comment (JonZeolla) [inline Taskfile.yml:142]: I have a similar but different fix in my PR among other standardization

https://github.com/SeisoLLC/easy_infra/pull/279/files

I'd undo this and just rebase on `main` after my fix goes in.  It's needed to fix releasing anyway, so it needs to happen first

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:99]: nit: make absolute and handle in one step

```suggestion
 && cd /tmp/git \
```

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:96]:
```suggestion
 && mkdir /tmp/git \
```

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:95]:
```suggestion
```

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:97]:
```suggestion
 && curl -L https://mirrors.edge.kernel.org/pub/software/scm/git/git-${GIT_VERSION#v}.tar.gz -o /tmp/git.tar.gz \
```

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:98]:
```suggestion
 && tar -zxf /tmp/git.tar.gz -C /tmp/git --strip-components=1 \
```

File: build/Dockerfile.base
Comment (JonZeolla) [inline build/Dockerfile.base:100]:
```suggestion
 && cd .. \
```

File: easy_infra.yml
Comment (JonZeolla) [inline easy_infra.yml:79]: This will ensure that `git --version` is run and successful in the correct images, due to this:

https://github.com/SeisoLLC/easy_infra/blob/971a571328b29a8254de1bf0f5f0b1e158508fc4/tests/test.py#L63-L149

One example [here](https://github.com/SeisoLLC/easy_infra/actions/runs/5958880313/job/16163772092#step:13:70)

But it lacks other useful tests of the logging end-to-end.

I want to forego that for the time being, because we'll be refactoring this all soon to move to otel, but can you please put this on the backlog and share the task link?  Specifically we need to add tests of easy_infra log generation.  This is one that we will likely implement using goss and start running the different testing frameworks in parallel

**Review by JonZeolla — CHANGES_REQUESTED** [review]

File: docs/Technical Details/index.rst
Comment (JonZeolla) [reply docs/Technical Details/index.rst:203]: We should also be very clear that this only applies when the dir in the container with the git repo which contains iac is not the working dir

File: docs/Technical Details/index.rst
Comment (JonZeolla) [reply docs/Technical Details/index.rst:193]: I turned on building this feature branch temporarily so you can work through it

https://easy-infra.readthedocs.io/en/chore-update-ubuntu-and-add-var/

File: docs/Logging/index.rst
Comment (ChiefHolland) [inline docs/Logging/index.rst:50]:
```suggestion

Mounted directories must be considered "safe" by git, in order for logging to function properly.
```

**Review by JonZeolla — APPROVED** [review]
🚀

**Pull request merged**
